crowbar_framework/app/models/service_object.rb

```diff
@@ -822,8 +822,19 @@ def apply_role(role, inst, in_queue)
     # We also check that all nodes we'll require are in the ready state.
     #
 
-    # Initialize variables used in ensure at the end of the method
-    chef_daemon_nodes = []
+    nodes_without_admin = NodeObject.find_all_nodes.reject do |node|
+      node[:platform] == "windows" || node.admin?
+    end.map(&:name)
+
+    apply_locks = nodes_without_admin.map do |node|
```

Comment:
Hrm. Are we really locking absolutely all nodes, even if they're not related to the proposal being applied?